File: sdk/go/generate_cli_docs.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

type CommandDoc struct {
	Path        string
	Title       string
	Description string
	Synopsis    string
	Options     []Option
	Examples    []string
}

type Option struct {
	Long     string
	Short    string
	Desc     string
	Default  string
}

func main() {
	commands := []CommandDoc{
		// Development Commands
		{
			Path:        "commands/dev/serve.md",
			Title:       "tsk serve",
			Description: "Start development server for local development and testing",
			Synopsis:    "tsk serve [OPTIONS] [port]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--config", "-c", "Configuration file path", "peanu.peanuts"},
				{"--hot-reload", "-r", "Enable hot reloading", "false"},
				{"--host", "", "Server host address", "localhost"},
				{"--timeout", "-t", "Request timeout in seconds", "30"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk serve 8080",
				"tsk serve --hot-reload 3000",
				"tsk serve --config dev.peanuts 9000",
			},
		},
		{
			Path:        "commands/dev/compile.md",
			Title:       "tsk compile",
			Description: "Compile TuskLang files to binary format for production",
			Synopsis:    "tsk compile [OPTIONS] <file>",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--output", "-o", "Output file path", "-"},
				{"--optimize", "", "Enable optimization", "false"},
				{"--format", "-f", "Output format (binary, json)", "binary"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk compile app.tsk",
				"tsk compile -o app.bin app.tsk",
				"tsk compile --optimize config.tsk",
			},
		},
		{
			Path:        "commands/dev/optimize.md",
			Title:       "tsk optimize",
			Description: "Optimize TuskLang files for better performance",
			Synopsis:    "tsk optimize [OPTIONS] <file>",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--output", "-o", "Output file path", "-"},
				{"--level", "-l", "Optimization level (1-3)", "2"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk optimize config.tsk",
				"tsk optimize -o optimized.tsk config.tsk",
				"tsk optimize --level 3 app.tsk",
			},
		},
		// Testing Commands
		{
			Path:        "commands/test/all.md",
			Title:       "tsk test all",
			Description: "Run all test suites including parser, FUJSEN, SDK, and performance tests",
			Synopsis:    "tsk test all [OPTIONS]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--parallel", "-p", "Run tests in parallel", "false"},
				{"--timeout", "-t", "Test timeout in seconds", "300"},
				{"--coverage", "-c", "Generate coverage report", "false"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk test all",
				"tsk test all --parallel",
				"tsk test all --coverage",
			},
		},
		{
			Path:        "commands/test/parser.md",
			Title:       "tsk test parser",
			Description: "Run parser-specific tests to validate TuskLang syntax parsing",
			Synopsis:    "tsk test parser [OPTIONS]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--file", "-f", "Test specific file", "-"},
				{"--timeout", "-t", "Test timeout in seconds", "60"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk test parser",
				"tsk test parser --file test.tsk",
				"tsk test parser --verbose",
			},
		},
		{
			Path:        "commands/test/fujsen.md",
			Title:       "tsk test fujsen",
			Description: "Run FUJSEN intelligence system tests",
			Synopsis:    "tsk test fujsen [OPTIONS]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--suite", "-s", "Test suite name", "-"},
				{"--timeout", "-t", "Test timeout in seconds", "120"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk test fujsen",
				"tsk test fujsen --suite core",
				"tsk test fujsen --verbose",
			},
		},
		{
			Path:        "commands/test/sdk.md",
			Title:       "tsk test sdk",
			Description: "Run SDK integration and functionality tests",
			Synopsis:    "tsk test sdk [OPTIONS]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--module", "-m", "Test specific module", "-"},
				{"--timeout", "-t", "Test timeout in seconds", "180"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk test sdk",
				"tsk test sdk --module peanut",
				"tsk test sdk --verbose",
			},
		},
		{
			Path:        "commands/test/performance.md",
			Title:       "tsk test performance",
			Description: "Run performance benchmarks and load tests",
			Synopsis:    "tsk test performance [OPTIONS]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--duration", "-d", "Test duration in seconds", "60"},
				{"--concurrent", "-c", "Number of concurrent requests", "10"},
				{"--output", "-o", "Output file for results", "-"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk test performance",
				"tsk test performance --duration 120",
				"tsk test performance --concurrent 50",
			},
		},
		// Configuration Commands
		{
			Path:        "commands/config/get.md",
			Title:       "tsk config get",
			Description: "Get configuration value using dot notation path",
			Synopsis:    "tsk config get [OPTIONS] <key.path> [dir]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--default", "-d", "Default value if key not found", "-"},
				{"--type", "-t", "Expected value type", "auto"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk config get server.port",
				"tsk config get database.host .",
				"tsk config get app.name --default 'My App'",
			},
		},
		{
			Path:        "commands/config/check.md",
			Title:       "tsk config check",
			Description: "Check configuration hierarchy and file loading order",
			Synopsis:    "tsk config check [OPTIONS] [path]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--detailed", "-d", "Show detailed information", "false"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk config check",
				"tsk config check .",
				"tsk config check --detailed",
			},
		},
		{
			Path:        "commands/config/validate.md",
			Title:       "tsk config validate",
			Description: "Validate configuration syntax and structure",
			Synopsis:    "tsk config validate [OPTIONS] [path]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--schema", "-s", "Schema file for validation", "-"},
				{"--strict", "", "Enable strict validation", "false"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk config validate",
				"tsk config validate .",
				"tsk config validate --schema schema.json",
			},
		},
		{
			Path:        "commands/config/compile.md",
			Title:       "tsk config compile",
			Description: "Compile configuration files to binary format",
			Synopsis:    "tsk config compile [OPTIONS] [path]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--output", "-o", "Output directory", "-"},
				{"--format", "-f", "Output format (pnt, json)", "pnt"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk config compile",
				"tsk config compile .",
				"tsk config compile --output dist/",
			},
		},
		// Peanuts Commands
		{
			Path:        "commands/peanuts/compile.md",
			Title:       "tsk peanuts compile",
			Description: "Compile .peanuts or .tsk files to binary .pnt format",
			Synopsis:    "tsk peanuts compile [OPTIONS] <file>",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--output", "-o", "Output file path", "-"},
				{"--optimize", "", "Enable optimization", "false"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk peanuts compile config.peanuts",
				"tsk peanuts compile -o config.pnt config.tsk",
				"tsk peanuts compile --optimize app.peanuts",
			},
		},
		{
			Path:        "commands/peanuts/load.md",
			Title:       "tsk peanuts load",
			Description: "Load and display binary peanuts file contents",
			Synopsis:    "tsk peanuts load [OPTIONS] <file.pnt>",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--format", "-f", "Output format (table, json, yaml)", "table"},
				{"--key", "-k", "Show specific key only", "-"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk peanuts load config.pnt",
				"tsk peanuts load --format json config.pnt",
				"tsk peanuts load --key server.port config.pnt",
			},
		},
		// Utility Commands
		{
			Path:        "commands/utility/parse.md",
			Title:       "tsk parse",
			Description: "Parse TuskLang file and display syntax tree",
			Synopsis:    "tsk parse [OPTIONS] <file>",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--format", "-f", "Output format (tree, json, yaml)", "tree"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk parse config.tsk",
				"tsk parse --format json app.tsk",
				"tsk parse --verbose schema.tsk",
			},
		},
		{
			Path:        "commands/utility/validate.md",
			Title:       "tsk validate",
			Description: "Validate TuskLang file syntax and structure",
			Synopsis:    "tsk validate [OPTIONS] <file>",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--schema", "-s", "Schema file for validation", "-"},
				{"--strict", "", "Enable strict validation", "false"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk validate config.tsk",
				"tsk validate --schema schema.json app.tsk",
				"tsk validate --strict schema.tsk",
			},
		},
		{
			Path:        "commands/utility/version.md",
			Title:       "tsk version",
			Description: "Display TuskLang CLI version and build information",
			Synopsis:    "tsk version [OPTIONS]",
			Options: []Option{
				{"--help", "-h", "Show help for this command", "-"},
				{"--json", "-j", "Output in JSON format", "false"},
				{"--quiet", "-q", "Suppress output", "false"},
				{"--verbose", "-v", "Show verbose output", "false"},
			},
			Examples: []string{
				"tsk version",
				"tsk version --json",
				"tsk version --verbose",
			},
		},
	}

	baseDir := "cli-docs/go"

	for _, cmd := range commands {
		filePath := filepath.Join(baseDir, cmd.Path)
		dir := filepath.Dir(filePath)
		
		// Create directory if it doesn't exist
		if err := os.MkdirAll(dir, 0755); err != nil {
			fmt.Printf("Error creating directory %s: %v\n", dir, err)
			continue
		}

		content := generateCommandDoc(cmd)
		if err := os.WriteFile(filePath, []byte(content), 0644); err != nil {
			fmt.Printf("Error writing file %s: %v\n", filePath, err)
		} else {
			fmt.Printf("Created: %s\n", filePath)
		}
	}

	// Create README files for command categories
	categories := []string{
		"commands/test", "commands/services", "commands/cache", "commands/config",
		"commands/binary", "commands/peanuts", "commands/ai", "commands/css",
		"commands/license", "commands/utility", "commands/cache/memcached", "examples",
	}

	for _, category := range categories {
		readmePath := filepath.Join(baseDir, category, "README.md")
		dir := filepath.Dir(readmePath)
		
		if err := os.MkdirAll(dir, 0755); err != nil {
			fmt.Printf("Error creating directory %s: %v\n", dir, err)
			continue
		}

		content := generateCategoryREADME(category)
		if err := os.WriteFile(readmePath, []byte(content), 0644); err != nil {
			fmt.Printf("Error writing file %s: %v\n", readmePath, err)
		} else {
			fmt.Printf("Created: %s\n", readmePath)
		}
	}

	fmt.Println("\nCLI documentation generation completed!")
}

func generateCommandDoc(cmd CommandDoc) string {
	var optionsTable strings.Builder
	optionsTable.WriteString("| Option | Short | Description | Default |\n")
	optionsTable.WriteString("|--------|-------|-------------|---------|\n")
	
	for _, opt := range cmd.Options {
		optionsTable.WriteString(fmt.Sprintf("| %s | %s | %s | %s |\n", opt.Long, opt.Short, opt.Desc, opt.Default))
	}

	var examples strings.Builder
	for _, ex := range cmd.Examples {
		examples.WriteString(fmt.Sprintf("```bash\n%s\n```\n\n", ex))
	}

	return fmt.Sprintf("# %s\n\n%s\n\n## Synopsis\n\n```bash\n%s\n```\n\n## Description\n\n%s\n\n## Options\n\n%s\n\n## Examples\n\n%s\n\n## Go API Usage\n\n```go\npackage main\n\nimport (\n    \"fmt\"\n    \"log\"\n    \"github.com/tusklang/go-sdk/cli\"\n)\n\nfunc main() {\n    // Create CLI client\n    client := cli.NewClient()\n    \n    // Execute command\n    result, err := client.Execute(\"%s\")\n    if err != nil {\n        log.Fatal(err)\n    }\n    \n    fmt.Printf(\"Command executed successfully: %%s\\n\", result.Output)\n}\n```\n\n## Output\n\n### Success Output\n\nWhen the command completes successfully:\n\n- Success status with ✅ symbol\n- Command results and summary\n- Execution duration\n- Relevant metrics\n\n### Error Output\n\nWhen the command fails:\n\n- Error status with ❌ symbol\n- Detailed error message\n- Troubleshooting suggestions\n\n## Exit Codes\n\n| Code | Description |\n|------|-------------|\n| 0 | Success - Command completed |\n| 1 | Error - Command failed |\n| 2 | Error - Invalid arguments |\n| 3 | Error - Configuration error |\n\n## Related Commands\n\n- Related command links will be added here\n\n## Notes\n\n- **Configuration**: Uses settings from peanu.peanuts file\n- **Performance**: Optimized for Go applications\n- **Compatibility**: Works across all supported platforms\n\n## See Also\n\n- [Commands Overview](../README.md)\n- [Configuration Guide](../../../go/docs/PNT_GUIDE.md)\n- [Examples](../../examples/)\n", cmd.Title, cmd.Description, cmd.Synopsis, cmd.Description, optionsTable.String(), examples.String(), strings.Fields(cmd.Title)[1], cmd.Title)
}

func generateCategoryREADME(category string) string {
	title := strings.Title(strings.ReplaceAll(strings.TrimPrefix(category, "commands/"), "/", " "))
	
	return fmt.Sprintf(`# %s Commands

%s operations for TuskLang Go CLI.

## Available Commands

| Command | Description |
|---------|-------------|
| [Command documentation files will be listed here] | [Brief descriptions] |

## Common Use Cases

- **Use Case 1**: Description of common scenario
- **Use Case 2**: Description of common scenario
- **Use Case 3**: Description of common scenario

## Go-Specific Notes

The Go CLI %s commands provide:

- **Feature 1**: Description
- **Feature 2**: Description
- **Feature 3**: Description

## Examples

### Basic Usage

```bash
# Example command usage
tsk command example
```

### Advanced Usage

```bash
# Advanced command usage
tsk command --option value
```

## Related Categories

- [Other command categories](../README.md)
`, title, title, strings.ToLower(title))
} 

File: sdk/ruby/generate_cli_docs.rb
#!/usr/bin/env ruby

# Generate complete CLI documentation for TuskLang Ruby SDK
# This script creates all missing command documentation files

require 'fileutils'

# Command structure from CLI_DOCS_STRUCTURE_PROMPT.md
COMMANDS = {
  'db' => {
    'README.md' => 'Database operations for TuskLang Ruby SDK.',
    'status.md' => 'Check database connection status',
    'migrate.md' => 'Run migration file',
    'console.md' => 'Open interactive database console',
    'backup.md' => 'Backup database',
    'restore.md' => 'Restore from backup file',
    'init.md' => 'Initialize SQLite database'
  },
  'dev' => {
    'README.md' => 'Development tools for TuskLang Ruby SDK.',
    'serve.md' => 'Start development server',
    'compile.md' => 'Compile TuskLang files',
    'optimize.md' => 'Optimize code and configuration'
  },
  'test' => {
    'README.md' => 'Test execution for TuskLang Ruby SDK.',
    'all.md' => 'Run all tests',
    'parser.md' => 'Test TuskLang parser',
    'fujsen.md' => 'Test FUJSEN functions',
    'sdk.md' => 'Test SDK functionality',
    'performance.md' => 'Run performance tests'
  },
  'services' => {
    'README.md' => 'Service management for TuskLang Ruby SDK.',
    'start.md' => 'Start TuskLang services',
    'stop.md' => 'Stop TuskLang services',
    'restart.md' => 'Restart TuskLang services',
    'status.md' => 'Check service status'
  },
  'cache' => {
    'README.md' => 'Cache operations for TuskLang Ruby SDK.',
    'clear.md' => 'Clear all caches',
    'status.md' => 'Show cache status',
    'warm.md' => 'Warm up caches',
    'memcached' => {
      'status.md' => 'Check Memcached status',
      'stats.md' => 'Show Memcached statistics',
      'flush.md' => 'Flush Memcached cache',
      'restart.md' => 'Restart Memcached service',
      'test.md' => 'Test Memcached connection'
    }
  },
  'config' => {
    'README.md' => 'Configuration management for TuskLang Ruby SDK.',
    'get.md' => 'Get configuration value by path',
    'check.md' => 'Check configuration hierarchy',
    'validate.md' => 'Validate entire configuration chain',
    'compile.md' => 'Auto-compile all peanu.tsk files',
    'docs.md' => 'Generate configuration documentation',
    'clear-cache.md' => 'Clear configuration cache',
    'stats.md' => 'Show configuration performance statistics'
  },
  'binary' => {
    'README.md' => 'Binary compilation for TuskLang Ruby SDK.',
    'compile.md' => 'Compile to binary format',
    'execute.md' => 'Execute binary file',
    'benchmark.md' => 'Benchmark binary performance',
    'optimize.md' => 'Optimize binary size and speed'
  },
  'peanuts' => {
    'README.md' => 'Peanut configuration for TuskLang Ruby SDK.',
    'compile.md' => 'Compile Peanut configuration',
    'auto-compile.md' => 'Auto-compile Peanut files',
    'load.md' => 'Load Peanut configuration'
  },
  'ai' => {
    'README.md' => 'AI integrations for TuskLang Ruby SDK.',
    'claude.md' => 'Interact with Claude AI',
    'chatgpt.md' => 'Interact with ChatGPT',
    'custom.md' => 'Use custom AI provider',
    'config.md' => 'Configure AI settings',
    'setup.md' => 'Setup AI API keys',
    'test.md' => 'Test AI connections',
    'complete.md' => 'Complete code with AI',
    'analyze.md' => 'Analyze code with AI',
    'optimize.md' => 'Optimize code with AI',
    'security.md' => 'Security analysis with AI'
  },
  'css' => {
    'README.md' => 'CSS utilities for TuskLang Ruby SDK.',
    'expand.md' => 'Expand CSS shorthand',
    'map.md' => 'Generate CSS source maps'
  },
  'license' => {
    'README.md' => 'License management for TuskLang Ruby SDK.',
    'check.md' => 'Check license status',
    'activate.md' => 'Activate license'
  },
  'utility' => {
    'README.md' => 'General utilities for TuskLang Ruby SDK.',
    'parse.md' => 'Parse TuskLang files',
    'validate.md' => 'Validate TuskLang syntax',
    'convert.md' => 'Convert between formats',
    'get.md' => 'Get value from configuration',
    'set.md' => 'Set value in configuration',
    'version.md' => 'Show version information',
    'help.md' => 'Show help information'
  }
}

def create_command_template(category, command, description)
  command_name = command.sub('.md', '')
  full_command = "tsk #{category} #{command_name}"
  
  <<~TEMPLATE
# #{full_command}

#{description}

## Synopsis

```bash
#{full_command} [options]
```

## Description

#{description.capitalize} for the TuskLang Ruby SDK.

## Options

| Option | Short | Description | Default |
|--------|-------|-------------|---------|
| --help | -h | Show help for this command | - |

## Arguments

No arguments required.

## Examples

### Basic Usage
```bash
# #{description.downcase}
#{full_command}
```

**Output:**
```
✅ Command executed successfully
```

## Ruby API Usage

```ruby
# Programmatic usage
require 'tusk_lang'

# Example implementation
def #{command_name.gsub('-', '_')}
  puts "Executing #{command_name}..."
  # Implementation here
end
```

## Output

### Success Output
```
✅ Command executed successfully
```

### Error Output
```
❌ Command failed: [error message]
```

## Exit Codes

| Code | Description |
|------|-------------|
| 0 | Success - Command executed |
| 1 | Error - Command failed |

## Related Commands

- [tsk #{category}](./README.md) - #{category.capitalize} commands overview

## Notes

- This is a placeholder for the #{command_name} command
- Replace with actual implementation details
- Add Ruby-specific examples and integration patterns

## See Also

- [#{category.capitalize} Commands Overview](./README.md)
- [Examples](../../examples/basic-usage.md)
TEMPLATE
end

def create_category_readme(category, description)
  commands = COMMANDS[category].keys.reject { |k| k == 'README.md' }
  
  command_links = commands.map do |cmd|
    cmd_name = cmd.sub('.md', '')
    "- [tsk #{category} #{cmd_name}](./#{cmd}) - #{COMMANDS[category][cmd]}"
  end.join("\n")
  
  <<~TEMPLATE
# #{category.capitalize} Commands

#{description}

## Available Commands

#{command_links}

## Common Use Cases

- **Development**: Common development scenarios
- **Production**: Production deployment patterns
- **Debugging**: Troubleshooting and debugging
- **Integration**: Framework and tool integration

## Ruby Specific Notes

- Follows Ruby conventions and best practices
- Integrates with Rails, Jekyll, and other Ruby frameworks
- Provides Ruby API examples for programmatic usage
- Supports Ruby-specific configuration patterns

## Examples

```bash
# Basic usage examples
tsk #{category} --help
```

## Integration with Rails

```ruby
# Rails integration example
class Application < Rails::Application
  # TuskLang #{category} integration
end
```
TEMPLATE
end

def create_file(path, content)
  FileUtils.mkdir_p(File.dirname(path))
  File.write(path, content)
  puts "Created: #{path}"
end

# Generate all command files
COMMANDS.each do |category, commands|
  commands.each do |command, description|
    if command == 'README.md'
      # Create category README
      path = "cli-docs/ruby/commands/#{category}/README.md"
      content = create_category_readme(category, description)
      create_file(path, content)
    elsif description.is_a?(Hash)
      # Handle nested commands (like cache/memcached)
      description.each do |subcommand, subdescription|
        path = "cli-docs/ruby/commands/#{category}/#{command}/#{subcommand}"
        content = create_command_template("#{category} #{command}", subcommand, subdescription)
        create_file(path, content)
      end
    else
      # Create individual command file
      path = "cli-docs/ruby/commands/#{category}/#{command}"
      content = create_command_template(category, command, description)
      create_file(path, content)
    end
  end
end

# Create additional documentation files
additional_files = {
  'cli-docs/ruby/examples/README.md' => <<~CONTENT,
# Examples

Complete working examples for TuskLang Ruby SDK.

## Available Examples

- [Basic Usage](./basic-usage.md) - Common usage patterns
- [Workflows](./workflows.md) - Complete workflows
- [Integrations](./integrations.md) - Framework integrations

## Getting Started

Start with the basic usage examples and work your way up to complex integrations.
CONTENT

  'cli-docs/ruby/examples/basic-usage.md' => <<~CONTENT,
# Basic Usage Examples

Common usage patterns for TuskLang Ruby SDK.

## Configuration Management

```bash
# Get configuration value
tsk config get server.port

# Check configuration
tsk config check

# Validate configuration
tsk config validate
```

## Database Operations

```bash
# Initialize database
tsk db init

# Check status
tsk db status

# Run migration
tsk db migrate schema.sql
```

## Development Workflow

```bash
# Start server
tsk serve 3000

# Run tests
tsk test all

# Compile configuration
tsk config compile
```
CONTENT

  'cli-docs/ruby/examples/workflows.md' => <<~CONTENT,
# Complete Workflows

End-to-end workflows for TuskLang Ruby SDK.

## Development Workflow

1. **Initialize Project**
   ```bash
   tsk db init
   tsk config check
   ```

2. **Development**
   ```bash
   tsk serve 3000
   tsk test all
   ```

3. **Deployment**
   ```bash
   tsk config compile
   tsk db backup
   ```

## Testing Workflow

1. **Run Tests**
   ```bash
   tsk test all
   tsk test parser
   tsk test performance
   ```

2. **Debug Issues**
   ```bash
   tsk db console
   tsk config validate
   ```
CONTENT

  'cli-docs/ruby/examples/integrations.md' => <<~CONTENT,
# Framework Integrations

Integration examples for popular Ruby frameworks.

## Rails Integration

```ruby
# config/application.rb
class Application < Rails::Application
  # Load TuskLang configuration
  config_loader = PeanutConfig.new
  app_config = config_loader.load(Rails.root.join('config'))
  
  # Apply configuration
  config.app_name = app_config['app']['name']
  config.debug_mode = app_config['app']['debug']
end
```

## Jekyll Integration

```ruby
# _plugins/tusk_lang.rb
Jekyll::Hooks.register :site, :after_init do |site|
  config = TuskLang::TSK.from_file('_config.tsk')
  site.config.merge!(config.to_hash)
end
```

## Sinatra Integration

```ruby
# app.rb
require 'sinatra'
require 'tusk_lang'

class MyApp < Sinatra::Base
  configure do
    config = TuskLang::TSK.from_file('config.tsk')
    set :port, config.get_value('server', 'port')
  end
end
```
CONTENT

  'cli-docs/ruby/troubleshooting.md' => <<~CONTENT,
# Troubleshooting

Common issues and solutions for TuskLang Ruby SDK.

## Common Issues

### Command Not Found
```bash
# Install the gem
gem install tusk_lang

# Verify installation
tsk --version
```

### Database Connection Failed
```bash
# Initialize database
tsk db init

# Check status
tsk db status
```

### Configuration Not Found
```bash
# Create configuration
echo '[app]' > peanu.peanuts
echo 'name: "My App"' >> peanu.peanuts

# Check configuration
tsk config check
```

### Permission Denied
```bash
# Fix permissions
sudo gem install tusk_lang
```

## Getting Help

```bash
# Show help
tsk --help

# Command-specific help
tsk help [command]

# Version information
tsk --version
```
CONTENT
}

additional_files.each do |path, content|
  create_file(path, content)
end

puts "\n✅ CLI documentation generation complete!"
puts "Generated #{COMMANDS.values.flatten.count} command files"
puts "Generated #{additional_files.count} additional documentation files" 